### PR TITLE
Fix: Better handle database failures in session storage adapters

### PR DESCRIPTION
**Pull request by andy-liuu**
TODO:
- [x] Try it on a real app
- [x] Add unit tests
    - [x] Ready cases
    - [x] Not ready cases
- [x] Changeset message for a minor
- [x] Verify that the typescripting is ok
- [x] Create an issue to add method to all other session storage implementations and remove prisma's special treatment
- [x] Make it cleaner
- [x] Self review

### WHY are these changes introduced?

Fixes https://github.com/Shopify/first-party-library-planning/issues/682

## Type of change

- [ ] Patch: Bug (non-breaking change which fixes an issue)
- [x] Minor: New feature (non-breaking change which adds functionality)
- [ ] Major: Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have used `pnpm changeset` to create a draft changelog entry (do NOT update the `CHANGELOG.md` files manually)
- [x] I have added/updated tests for this change
- [ ] I have documented new APIs/updated the documentation for modified APIs (for public APIs)

### CODE REVIEW

File: packages/apps/session-storage/shopify-app-session-storage-prisma/src/__tests__/prisma.test.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import fs from 'fs';
+import * as fs from 'fs';
```
Comment (sle-c) [inline packages/apps/session-storage/shopify-app-session-storage-prisma/src/__tests__/prisma.test.ts:1]: do we need to turn this to import *?

File: packages/apps/session-storage/shopify-app-session-storage-prisma/src/prisma.ts [hunk 2]
```diff
@@ -129,13 +132,30 @@ export class PrismaSessionStorage<T extends PrismaClient>
     return sessions.map((session) => this.rowToSession(session));
   }
 
+  public async isReady(): Promise<boolean> {
```
[the remark this answers is not in the document]
Comment (sle-c) [reply packages/apps/session-storage/shopify-app-session-storage-prisma/src/prisma.ts:135]: what do you think about `isConnected()`?

File: packages/apps/session-storage/shopify-app-session-storage-prisma/src/prisma.ts [hunk 3]
```diff
@@ -129,13 +132,30 @@ export class PrismaSessionStorage<T extends PrismaClient>
...
+    try {
+      await this.pollForTable();
```
Comment (sle-c) [inline packages/apps/session-storage/shopify-app-session-storage-prisma/src/prisma.ts:137]: if `this.pollForTable()` catches all errors inside and we're catching all errors here, how do we know if the query succeeded?

File: packages/apps/session-storage/shopify-app-session-storage-prisma/src/prisma.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (andy-liuu) [reply packages/apps/session-storage/shopify-app-session-storage-prisma/src/prisma.ts:135]: `isConnected` is ok with me. Will change

File: packages/apps/session-storage/shopify-app-session-storage-prisma/src/prisma.ts (hunk 3 above)
Comment (sle-c) [reply packages/apps/session-storage/shopify-app-session-storage-prisma/src/prisma.ts:137]: my bad, I missed the `throw` statement at the end of `pollForTable`

File: packages/apps/session-storage/shopify-app-session-storage-prisma/src/prisma.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (andy-liuu) [reply packages/apps/session-storage/shopify-app-session-storage-prisma/src/prisma.ts]: Made a commit, but keep in mind that all of our session storage implementations will need this method eventually, and `isConnected` doesn't really make sense for `MemorySessionStorage`

**Pull request merged**